scalafix-core/src/main/scala/scalafix/rewrite/NoExtendsApp.scala

```diff
@@ -0,0 +1,124 @@
+package scalafix
+package rewrite
+
+import scala.meta.{Symbol => _, _}
+import scalafix.syntax._
+import scalafix.util.{Whitespace => _, _}
+import scala.meta.contrib._
+import scala.meta.tokens.Token._
+
+//TODO(gabro): move the helpers somewhere else
+import NoExtendsAppSyntax._
+
+case class NoExtendsApp(mirror: Mirror) extends SemanticRewrite(mirror) {
+  override def rewrite(ctx: RewriteCtx): Patch = {
+    def wrapBodyInMain(template: Template) =
+      ctx
+        .templateBodyTokens(template)
+        .map { body =>
+          val bodyIndentation = ctx.indentationTokens(body)
+          val open =
+            ctx.addLeft(
+              body.head,
+              s"\n${bodyIndentation}def main(args: Array[String]) = {")
+          val close = ctx.addRight(body.last, s"${bodyIndentation}}\n")
+          open + ctx.indent(body, bodyIndentation) + close
+        }
+        .asPatch
+
+    ctx.tree.collect {
+      case t: Defn.Object =>
+        ctx.removeParentFromTemplate("_root_.scala.App.", t.templ) +
+          wrapBodyInMain(t.templ)
+    }.asPatch
+  }
+}
+
+object NoExtendsAppSyntax {
+  implicit class XtensionRewriteCtx(ctx: RewriteCtx) {
```

Comment:
Great. Let's keep everything in this file then since moving things around will be a binary breaking change, I can do the migration for 0.5 https://github.com/scalacenter/scalafix/issues/193